### PR TITLE
Prevents users from specifying a server managed property as the objec…

### PR DESCRIPTION
**Pull request by dbernstein**
…t of an

ldp:hasMemberRelation predicate.

Resolves: https://jira.duraspace.org/browse/FCREPO-2519

# Notes
This pull request to my knowledge does not introduce breaking changes.

### CODE REVIEW

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java [hunk 1]
```diff
@@ -636,9 +642,38 @@ protected void replaceResourceWithStream(final FedoraResource resource,
             throw new RepositoryRuntimeException(e);
         }
 
+        ensureConsistencyOfInput(inputModel);
+
         resource.replaceProperties(translator(), inputModel, resourceTriples);
     }
 
+    private void ensureConsistencyOfInput(final Model inputModel) throws BadRequestException {
+        LOGGER.info("ensuring consistency!");
```
Comment (escowles) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: Can you trim down this logging? I think `info` is too much here, and we probably want only `debug` if there is no exception.

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java [hunk 2]
```diff
@@ -636,9 +642,38 @@ protected void replaceResourceWithStream(final FedoraResource resource,
...
+        LOGGER.info("iterator.hasNext()={}", iterator.hasNext());
+
+        while (iterator.hasNext()) {
+            final Statement s = iterator.next();
+            LOGGER.debug("statement: s={}, p={}, o={}", s.getSubject(), s.getPredicate(), s.getObject());
+            if (s.getPredicate().equals(HAS_MEMBER_RELATION)) {
+                final RDFNode obj = s.getObject();
+                if (obj.isResource()) {
+                    final String uri = obj.asResource().getURI();
+                    RdfLexicon.serverManagedProperties.forEach(p -> {
```
Comment (escowles) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: I suspect there is a snappy one-liner using the Predicate, but can't this at least be shortened to: `if (RdfLexicon.serverManagedProperties.contains(p)) ...`

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java (hunk 1 above)
Comment (dbernstein) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: good catch.

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java (hunk 2 above)
Comment (awoods) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: Also, we want to be checking more than `serverManagedProperties`... which only includes:
```
    public static final Set<Property> serverManagedProperties = of(
            CREATED_DATE, CREATED_BY, LAST_MODIFIED_DATE, LAST_MODIFIED_BY);
```

**Comment by awoods** [comment]
@dbernstein : here is a pull on pull:
https://github.com/dbernstein/fcrepo4/pull/3

**Comment by awoods** [comment]
@escowles : If you are good with this PR, please feel free to squash/merge.

**Pull request merged**